### ISSUE TITLE
Windows Detections

### ISSUE THREAD

**Issue by varysuzuki** [UnamSanctam/UnamBinder#42]
some streamer opened my binded file, the problem was that it seemed that it didnt drop the real app and open it, i guess due to detections? or what could have caused that, he managed to get it to work but he had to open it like 7 times. also what could i do to make it more stabily drop and open process? since it doesnt seem to work or isnt stable this way

**Comment by UnamSanctam** [comment]
Could be detections, do you have "Add Windows Defender Exclusions" enabled? I haven't ever really experienced any instability, if I open a bound file with 2 files in it a hundred times then it opens the files correctly every time. Also, if you have a "Start Delay" enabled then there will be a delay. Can you press "Save" in the builder and then zip the .xml file and send it here?

**Comment by shainodev** [comment]
> Could be detections, do you have "Add Windows Defender Exclusions" enabled? I haven't ever really experienced any instability, if I open a bound file with 2 files in it a hundred times then it opens the files correctly every time. Also, if you have a "Start Delay" enabled then there will be a delay. Can you press "Save" in the builder and then zip the .xml file and send it here?

i think the reason was i modified it a bit, varysuzuki is my other acc. i had add windows defender exclusions enabled for temp and the place i dropped the miner. the stealer screenshoted it and i could see it live that he had to open it 7 times the stealer and miner opened ig but the real app didnt.
![image](https://user-images.githubusercontent.com/77803435/183899056-4fd73c3e-ba82-4876-a564-1c693bb51908.png)
these were the settings
![image](https://user-images.githubusercontent.com/77803435/183899168-e186ad96-8082-4d53-940e-d47ccfc8ccfa.png)
and this is the code (i put startup here instead of using the miners startup)

**Comment by shainodev** [comment]
the thing is i dont care if the executable is detected i want it to open the app without antivirus kicking in. could i do it like this, i use runpe to inject the stealer, and make the miner just be dropped to a wd excluded path and add it to startup with a command? and then idk how im going to inject or assembly.load the python file thats converted to exe (whats the real app) since it isnt c# so how should i do this? could i inject it into the application itself?

**Comment by shainodev** [comment]
![image](https://user-images.githubusercontent.com/77803435/183903319-40a5be17-2460-401e-adc8-ec3ce3dbb5e0.png)
![image](https://user-images.githubusercontent.com/77803435/183903353-28b4ed31-0e32-48fd-b518-d26f4028772f.png)
these are the detections of the stealer and miner. could the problem be that the tool itself is detected and then windows defender ignores the exclusion?

**Comment by UnamSanctam** [comment]
> i think the reason was i modified it a bit, varysuzuki is my other acc. i had add windows defender exclusions enabled for temp and the place i dropped the miner. the stealer screenshoted it and i could see it live that he had to open it 7 times the stealer and miner opened ig but the real app didnt.

Is "Run as Administrator" supposed to be disabled? Or are you running it as administrator in some other way?

> the thing is i dont care if the executable is detected i want it to open the app without antivirus kicking in. could i do it like this, i use runpe to inject the stealer, and make the miner just be dropped to a wd excluded path and add it to startup with a command? and then idk how im going to inject or assembly.load the python file thats converted to exe (whats the real app) since it isnt c# so how should i do this? could i inject it into the application itself?

Which antiviruses runtime scans are you trying to avoid, only Windows Defender or all of them? You should be able to inject a converted python file since it's most likely a native file.

>these are the detections of the stealer and miner. could the problem be that the tool itself is detected and then windows defender ignores the exclusion?

Well antiscan and other no-distribute scanners don't do all scans such as "Cloud Protection" since using them would distribute the file so they're not entirely accurate. Windows Defender usually doesn't ignore exclusions with the exception of being Assembly.Load (because it's in memory) though that can be bypassed by using an AMSI bypass so that shouldn't really be a problem.

**Comment by shainodev** [comment]
> Is "Run as Administrator" supposed to be disabled? Or are you running it as administrator in some other way?
Yeah I manually put <requestedExecutionLevel level="requireAdministrator" uiAccess="false"></requestedExecutionLevel> to the manifest file
> Which antiviruses runtime scans are you trying to avoid, only Windows Defender or all of them? You should be able to inject a converted python file since it's most likely a native file.
I mean 95% have windows defender so im trying to avoid that. Yeah but how do i inject it without them seeing its in another location or something? I want them to see that its the app that they opened
> Well antiscan and other no-distribute scanners don't do all scans such as "Cloud Protection" since using them would distribute the file so they're not entirely accurate. Windows Defender usually doesn't ignore exclusions with the exception of being Assembly.Load (because it's in memory) though that can be bypassed by using an AMSI bypass so that shouldn't really be a problem.
Im going to make the binder c# then.

**Comment by shainodev** [comment]
![image](https://user-images.githubusercontent.com/77803435/183944357-286af259-1116-474e-a11f-a7704f3bef27.png)
this is currently the code, i added the miner and stealer manually to the resources so i can just inject them but then id have to use Applaunch from the runtime directory, that will make it really detected right, no? cause im worried if i make it shellcode it'll be alot more detected since shellcode yeah

**Comment by UnamSanctam** [comment]
> I mean 95% have windows defender so im trying to avoid that. Yeah but how do i inject it without them seeing its in another location or something? I want them to see that its the app that they opened

If you inject it then they will see the file that you injected into, since that's the point of injection. You could of course create a "fake" file in the real location and inject into that but that would be a bit reduntant.

>this is currently the code, i added the miner and stealer manually to the resources so i can just inject them but then id have to use Applaunch from the runtime directory, that will make it really detected right, no? cause im worried if i make it shellcode it'll be alot more detected since shellcode yeah

If you inject something there it shouldn't really get detected by Windows Defender.

**Comment by shainodev** [comment]
> > I mean 95% have windows defender so im trying to avoid that. Yeah but how do i inject it without them seeing its in another location or something? I want them to see that its the app that they opened
> 
> If you inject it then they will see the file that you injected into, since that's the point of injection. You could of course create a "fake" file in the real location and inject into that but that would be a bit reduntant.
> 
> > this is currently the code, i added the miner and stealer manually to the resources so i can just inject them but then id have to use Applaunch from the runtime directory, that will make it really detected right, no? cause im worried if i make it shellcode it'll be alot more detected since shellcode yeah
> 
> If you inject something there it shouldn't really get detected by Windows Defender.

so how could i do this? i got an idea should i make it do all the powershell amsi shit and inject the stealer. miner (hidden of course) and then finally inject the real app into application.executablepath and close itself. but wouldnt that make it open in like 6 seconds? i dont wanna make it inject the real app first and then do the shit cause then if the miner and stealer injection part will get detected and deleted then the real app would still work and then they wont disable their av for it cause it already works and then i wont get anything

**Comment by shainodev** [comment]
is it bad to cipherreplace the #RESPARENT with parentid instead of just replacing it

**Comment by shainodev** [comment]
also do i have to do 
RuntimeEnvironment.GetRuntimeDirectory().Replace(""Framework64"", ""Framework"")
instead of just
RuntimeEnvironment.GetRuntimeDirectory()

**Comment by shainodev** [comment]
> > I mean 95% have windows defender so im trying to avoid that. Yeah but how do i inject it without them seeing its in another location or something? I want them to see that its the app that they opened
> 
> If you inject it then they will see the file that you injected into, since that's the point of injection. You could of course create a "fake" file in the real location and inject into that but that would be a bit reduntant.
> 
> > this is currently the code, i added the miner and stealer manually to the resources so i can just inject them but then id have to use Applaunch from the runtime directory, that will make it really detected right, no? cause im worried if i make it shellcode it'll be alot more detected since shellcode yeah
> 
> If you inject something there it shouldn't really get detected by Windows Defender.

and also if i inject the real app inside the binded.exe then it would only show the real app and not do anything that the binded app would've done right? so when i would inject the real app inside Application.Executablepath it wouldnt just do all the commands again and inject again, inevitably just making it spam open/inject the real app and it would lag their pc out

**Comment by shainodev** [comment]
i think i know what happened. in ur binder, the wd exclusion does this, open "powershell" and the argument "-EncodedCommand 'blablabla'" but it only works with "powershell", "powershell -EncodedCommand 'blabla'" i just tested it

**Comment by shainodev** [comment]
the app crashes when it tries to inject do u know what could cause that?
![image](https://user-images.githubusercontent.com/77803435/183988097-48ccbf07-7156-4a3d-a9fa-90b24d1f67d6.png)
the above one when i try to open it it says application failed to start, and the below one opens applaunch but with werfault.exe and closes itself

**Comment by shainodev** [comment]
nvm i realized i cant inject the converted python file to the C# binded file anyway, idk how to do this

**Comment by UnamSanctam** [comment]
> so how could i do this? i got an idea should i make it do all the powershell amsi shit and inject the stealer. miner (hidden of course) and then finally inject the real app into application.executablepath and close itself. but wouldnt that make it open in like 6 seconds? i dont wanna make it inject the real app first and then do the shit cause then if the miner and stealer injection part will get detected and deleted then the real app would still work and then they wont disable their av for it cause it already works and then i wont get anything

Well injection is nearly instant so injecting the three programs shouldn't even take a second. 

> is it bad to cipherreplace the #RESPARENT with parentid instead of just replacing it

Doesn't really matter usually, but you can do that if you want.

> and also if i inject the real app inside the binded.exe then it would only show the real app and not do anything that the binded app would've done right? so when i would inject the real app inside Application.Executablepath it wouldnt just do all the commands again and inject again, inevitably just making it spam open/inject the real app and it would lag their pc out

I'm not fully sure how your entire flow is set up, but I don't really understand what you're saying here.

> i think i know what happened. in ur binder, the wd exclusion does this, open "powershell" and the argument "-EncodedCommand 'blablabla'" but it only works with "powershell", "powershell -EncodedCommand 'blabla'" i just tested it

How it is should work fine, I just tested it like it currently is myself and it worked well for me on both my main computer and VM.

> is it a better idea to inject the shellcode of the stealer to conhost or to inject the stealer to applaunch

For Windows Defender it shouldn't really matter, it doesn't generally detect injections.

**Comment by shainodev** [comment]
> How it is should work fine, I just tested it like it currently is myself and it worked well for me on both my main computer and VM.

thats weird cause it will only work with powershell -EncodedCommand for me on Windows 11 Home
> > is it a better idea to inject the shellcode of the stealer to conhost or to inject the stealer to applaunch
> 
> For Windows Defender it shouldn't really matter, it doesn't generally detect injections.

im having problems the real app just wont inject for some reason
![image](https://user-images.githubusercontent.com/77803435/184003731-93eaea3d-554e-4377-99bb-0e8de98b3586.png)
what could be wrong

**Comment by UnamSanctam** [comment]
Is it a native application? And is it 32-bit or 64-bit?

**Comment by shainodev** [comment]
> Is it a native application? And is it 32-bit or 64-bit?

it was C and then C# and now i tried using shellcode of the C#, what has Any CPU and i used amd64+(x86) on donut

**Comment by UnamSanctam** [comment]
The injector (if you took the one from the miner) is only for 64-bit so make sure that it's built as 64-bit.

**Comment by shainodev** [comment]
> The injector (if you took the one from the miner) is only for 64-bit so make sure that it's built as 64-bit.

so i have to build the file as 64bit or just only convert it with 64bit on donut? or both

**Comment by UnamSanctam** [comment]
Both probably.

**Comment by shainodev** [comment]
oh i get why i get an error jesus christ im retarded. i used runpe instead of shellcode method

**Comment by shainodev** [comment]
![image](https://user-images.githubusercontent.com/77803435/184008961-2254e308-2500-4815-97ea-c7bc416fa42b.png)
so right now this code seems fine? it gets to the last part in about 2-3 seconds so that shouldnt raise any suspicion. also should i make the real tools folder have DisableAV.bat or something? cause they are so dumb they dont even know how to do that

**Comment by UnamSanctam** [comment]
Yes that seems fine, and you can have a .bat file sure, it would probably help.

**Comment by varysuzuki** [comment]
> Yes that seems fine, and you can have a .bat file sure, it would probably help.

yo btw whats the progress with the clipper and the stealer?

**Comment by UnamSanctam** [comment]
> yo btw whats the progress with the clipper and the stealer?

Now I'm working on the miner again since I got some inspiration from my other projects, but the clipper is basically done though I've gotten quite a few requests not to release it because it would be bad for crypto, and that it's also more malicious in nature compared to everything else I have released.

**Comment by shainodev** [comment]
> Now I'm working on the miner again since I got some inspiration from my other projects, but the clipper is basically done though I've gotten quite a few requests not to release it because it would be bad for crypto, and that it's also more malicious in nature compared to everything else I have released.

release it its only for educational purposes and learning 😄

**Comment by shainodev** [comment]
![image](https://user-images.githubusercontent.com/77803435/184184961-3bbf65fb-bc3e-48af-ab21-8d7219a63aed.png)
is this good? maybe it'll fool some dumb people who just know how to put a app into dnspy

**Comment by UnamSanctam** [comment]
Sure, looks good.

**Comment by shainodev** [comment]
> Sure, looks good.

is it bad to have all these try catches? should i for example remove the try catch inside the patch amsi and only have the try catch when i call patchamsi or will it cause it to get an exception and break something?

**Comment by UnamSanctam** [comment]
Since you have a try catch inside the functions then they're not really needed outside around the call of the function. It's better to have it inside the function usually since then you only have to do one.

**Comment by shainodev** [comment]
> Since you have a try catch inside the functions then they're not really needed outside around the call of the function. It's better to have it inside the function usually since then you only have to do one.

what about the run for example if gettheresource doesnt work or path.combine doesnt should i have try catch there?

**Comment by UnamSanctam** [comment]
Sure you can have them there, since you're calling multiple functions but otherwise for the most part you call functions without any arguments.

**Comment by shainodev** [comment]
> Sure you can have them there, since you're calling multiple functions but otherwise for the most part you call functions without any arguments.

wait does shellcode not work with c++ apps?

**Comment by shainodev** [comment]
yo and also some guy opened my stealer and the file searcher grabbed some "logs"? its luffich cloud free logs or something and idk what to do with them

**Comment by shainodev** [comment]
![image](https://user-images.githubusercontent.com/77803435/184205918-aa328a80-5215-4f0c-8099-5cd1e0ce92bf.png)
this happens when i try to inject the converted python with runpemethod to conhost

**Comment by shainodev** [comment]
now when i try injecting the converted python with runpemethod to conhost it works only for a second then closes

**Comment by shainodev** [comment]
hey i thought of making a c# dll that the tool includes in itself instead of a binder do u know how i can do this?

**Comment by UnamSanctam** [comment]
> now when i try injecting the converted python with runpemethod to conhost it works only for a second then closes

Does the converted python program work if you start it normally? 

> hey i thought of making a c# dll that the tool includes in itself instead of a binder do u know how i can do this?

Not sure what you mean, what would the C# DLL do and what tool would you include it in? If you want to include it in a C# program then just make it a resource and then load it, if it's for another language then there are other ways instead (such as using bin2h to make it into a header file for C/C++).

**Comment by varysuzuki** [comment]
> Not sure what you mean, what would the C# DLL do and what tool would you include it in? If you want to include it in a C# program then just make it a resource and then load it, if it's for another language then there are other ways instead (such as using bin2h to make it into a header file for C/C++).

i meant putting the stealer injection and miner drop in the C# dll itself and then calling the dll function from the "real app" that the victim will download, but idk how to do that without the dll getting flagged or the person thinking its a suspicious dll

**Comment by shainodev** [comment]
hey i made the dll work but i have a problem it gets detected by wd
![image](https://user-images.githubusercontent.com/77803435/184378979-6f7c0799-f961-4310-b992-a6483cfaddf6.png)
and idk what to do to fix that
this is the code
![image](https://user-images.githubusercontent.com/77803435/184379254-d3e24c6b-9f8d-46ea-899a-95f4ef20fbbc.png)
im thinking its cause of Convert.FromBase64String

**Comment by shainodev** [comment]
does everyone have C:\Users\Public?

**Comment by UnamSanctam** [comment]
Well, you can embed the DLL into the real app, and then add exclusions in the real app before loading the DLL (might have to use and AMSI bypass first).
